packages/slate-react/src/components/editable.tsx

```diff
@@ -550,13 +550,27 @@ export const Editable = (props: EditableProps) => {
           },
           [attributes.onCompositionEnd]
         )}
+        onCompositionUpdate={useCallback(
+          (event: React.CompositionEvent<HTMLDivElement>) => {
+            if (
+              hasEditableTarget(editor, event.target) &&
+              !isEventHandled(event, attributes.onCompositionUpdate)
+            ) {
+              state.isComposing = true
+            }
+          },
+          [attributes.onCompositionUpdate]
+        )}
         onCompositionStart={useCallback(
           (event: React.CompositionEvent<HTMLDivElement>) => {
             if (
               hasEditableTarget(editor, event.target) &&
               !isEventHandled(event, attributes.onCompositionStart)
             ) {
-              state.isComposing = true
+              const { selection } = editor
+              if (selection && Range.isExpanded(selection)) {
+                Editor.deleteFragment(editor)
```

Comment:
I think you should set selection to anchor after delete fragment.
Selection will be set to focus according [here](https://github.com/ianstormtaylor/slate/blob/master/packages/slate/src/transforms/text.ts#L181).  This causes the preview text to be misplaced when you typing.

Reply:
Thank you for the comment.
I don't think `Transforms.select(editor, { anchor, focus})` is necessary after delete fragment.
`Editor.deleteFragment(editor)` should set selection if it is needed.

You can test if fixed version works correctly or not from Netlify link below. If it doesn't work as you expected, please let me know.

![Screenshot from 2019-12-28 13-51-51](https://user-images.githubusercontent.com/29003390/71539071-2020f080-297a-11ea-9cc0-ea2ed4a1868c.png)

Reply:
You are right. Actually I had misplace problem in v0.56.0, and i fixed by that code. Your code do not have this problem. I think I‘ll figure it out later

Reply:
Chrome:81.0.4044.129
macOS:10.15.4 (19E266)
slate:0.58.0
The problem still exists！